cirq-core/cirq/sim/simulator_base.py

```diff
@@ -226,13 +286,43 @@ def _run(
 
         measurements: Dict[str, List[np.ndarray]] = {}
         for _ in range(repetitions):
+            copies = {a: a.copy() for a in set(act_on_args.values())}
```

Comment:
Why are copies required here? I don't see anywhere in `_core_iterator` that mutates the state, and these copies could be massively expensive for many-qubit states.

Reply:
Core iterator calls act-on which modifies the state. So this is an optimization versus running the prefix each time, which would be even more expensive than the copy. This copy has existed in some form or other since before I joined.

That said, there's an optimization opportunity here, in that the final repetition can use the original state instead of copying. Seems like small potatoes, but would make a notable difference when repetitions is 1. I'll make that change.